Assets/Scripts/Guns/EdgeGun.cs

```diff
@@ -20,7 +24,35 @@ public EdgeGun(GraphService graphService, Material edgeMaterial)
 
     public string GunName => "Edge";
 
+    private EdgeMode Mode
+    {
+      get => _mode;
+      set
+      {
+        if(_mode != value)
+        {
+          ClearPreviouslyHitNode();
+          _mode = value;
+        }
+      }
+    }
+
+    private void ClearPreviouslyHitNode()
+    {
+      if (previouslyHitNode != null)
+      {
+        previouslyHitNode.gameObject.DisableGlow();
+        previouslyHitNode = null;
+      }
+    }
+
     public void OnMoveDown(Transform playerTransform, Camera camera)
+    {
+      Mode = EdgeMode.Create;
+      ModifyEdge(camera);
+    }
+
+    private void ModifyEdge(Camera camera)
```

Comment:
I don't like the name of this function, it does not modify an edge. It's more like OnActonPerformed or something